File: MobilneSystemyInformatyczne/ShareWindsurfingEquipment/src/main/java/com/project/ShareWindsurfingEquipment/model/ItemRental.java
package com.project.ShareWindsurfingEquipment.model;

import javax.persistence.*;

@Entity
public class ItemRental {

    @EmbeddedId
    private ItemRentalKey id;

    @ManyToOne
    @MapsId("rentalId")
    @JoinColumn(name = "rental_id")
    private Rental rental;

    @ManyToOne
    @MapsId("equipmentId")
    @JoinColumn(name = "equipment_id")
    private Equipment equipment;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "rental_review_id", referencedColumnName = "id")
    private RentalReview rentalReview;

    public ItemRentalKey getId() {
        return id;
    }

    public void setId(ItemRentalKey id) {
        this.id = id;
    }

    public Rental getRental() {
        return rental;
    }

    public void setRental(Rental rental) {
        this.rental = rental;
    }

    public Equipment getEquipment() {
        return equipment;
    }

    public void setEquipment(Equipment equipment) {
        this.equipment = equipment;
    }

    public RentalReview getRentalReview() {
        return rentalReview;
    }

    public void setRentalReview(RentalReview rentalReview) {
        this.rentalReview = rentalReview;
    }

    @Override
    public String toString() {
        return "ItemRental{" +
                "id=" + id +
                ", rental=" + rental +
                ", equipment=" + equipment +
                ", rentalReview=" + rentalReview +
                '}';
    }
}

File: MSI-project/src/app/models/itemRental.ts
import { Equipment } from "./equipment";
import { ItemRentalKey } from "./itemRentalKey";
import { Rental } from "./rental";
import { RentalReview } from "./rentalReview";

export class ItemRental {

  constructor(
    private _id: ItemRentalKey,
    private _rental: Rental,
    private _equipement: Equipment,
    private _rentalReview: RentalReview
    ){}

    public get rentalReview(): RentalReview {
      return this._rentalReview;
    }
    public set rentalReview(value: RentalReview) {
      this._rentalReview = value;
    }
    public get equipement(): Equipment {
      return this._equipement;
    }
    public set equipement(value: Equipment) {
      this._equipement = value;
    }
    public get rental(): Rental {
      return this._rental;
    }
    public set rental(value: Rental) {
      this._rental = value;
    }
    public get id(): ItemRentalKey {
      return this._id;
    }
    public set id(value: ItemRentalKey) {
      this._id = value;
    }
}
